cmds/db-manager/main.go

```diff
@@ -0,0 +1,229 @@
+	}
+)
+
+func main() {
+	flag.Parse()
+	if *path == "" {
+		log.Panic("Must specify schemas_dir path")
+	}
+	if (*dbVersion == "" && *step == 0) || (*dbVersion != "" && *step != 0) {
+		log.Panic("Must specify a db_version or migration_step to goto")
+	}
+	if *dbVersion != "" && !semver.IsValid(*dbVersion) {
+		log.Panic("db_version must be in a valid format ex: v1.2.3")
+	}
+	uriParams := map[string]string{
+		"host":             *cockroachParams.host,
+		"port":             strconv.Itoa(*cockroachParams.port),
+		"user":             *cockroachParams.user,
+		"ssl_mode":         *cockroachParams.sslMode,
+		"ssl_dir":          *cockroachParams.sslDir,
+		"application_name": "SchemaManager",
+		"db_name":          filepath.Base(*path),
+	}
+	postgresURI, err := dssCockroach.BuildURI(uriParams)
+	if err != nil {
+		log.Panic("Failed to build URI", zap.Error(err))
+	}
+	myMigrater, err := New(*path, postgresURI)
+	if err != nil {
+		log.Panic(err)
+	}
+	defer func() {
+		if _, err := myMigrater.Close(); err != nil {
+			log.Println(err)
+		}
+	}()
+	preMigrationStep, _, err := myMigrater.Version()
+	if err != migrate.ErrNilVersion && err != nil {
+		log.Panic(err)
+	}
+	totalMoves, err := myMigrater.DoMigrate(*dbVersion, *step)
+	if err != nil {
+		log.Panic(err)
+	}
+	postMigrationStep, dirty, err := myMigrater.Version()
+	if err != nil {
+		log.Fatal("Failed to get Migration Step for confirmation")
+	}
+	if totalMoves == 0 {
+		log.Println("No Changes")
+	} else {
+		log.Printf("Moved %d step(s) in total from Step %d to Step %d", intAbs(totalMoves), preMigrationStep, postMigrationStep)
+	}
+
+	currentDBVersion, err := getCurrentDBVersion(postgresURI)
+	if err != nil {
+		log.Fatal("Failed to get Current DB version for confirmation")
+	}
+	log.Printf("DB Version: %s, Migration Step # %d, Dirty: %v", currentDBVersion, postMigrationStep, dirty)
+}
+
+// DoMigrate performs the migration given the desired state we want to reach
+func (m *MyMigrate) DoMigrate(desiredDBVersion string, desiredStep int) (int, error) {
+	totalMoves := 0
+	migrateDirection, err := m.MigrationDirection(desiredDBVersion, desiredStep)
+	if err != nil {
+		return totalMoves, err
+	}
+	for migrateDirection != 0 {
+		err = m.Steps(int(migrateDirection))
+		if err != nil {
+			return totalMoves, err
+		}
+		totalMoves += int(migrateDirection)
+		log.Printf("Migrated %s by %d step", migrateDirection.String(), intAbs(int(migrateDirection)))
+		migrateDirection, err = m.MigrationDirection(desiredDBVersion, *step)
+		if err != nil {
+			return totalMoves, err
+		}
+	}
+	return totalMoves, nil
+}
+
+// New instanciates a new migrate object
+func New(path string, dbURI string) (*MyMigrate, error) {
+	crdbURI := strings.Replace(dbURI, "postgresql", "cockroachdb", 1)
+	path = fmt.Sprintf("file://%v", path)
+	migrater, err := migrate.New(path, crdbURI)
```

Comment:
Either migrater should be valid or err should be != nil; both of those conditions shouldn't happen at the same time.  So, we should immediately return nil, err if err != nil here and we should return myMigrater, nil at the end of the function.